lib/rules/no-useless-escape.js

```diff
@@ -23,6 +23,12 @@ const VALID_STRING_ESCAPES = [
     "\r"
 ];
 
+const VALID_TEMPLATE_ESCAPES = VALID_STRING_ESCAPES.concat([
+    "$",
+    "{",
+    "}"
```

Comment:
I was thinking of a case where `}` is present in a template literal, but playing around with it some more, I think you're right. Will update!